### PR TITLE
2.0.48 bug: AttributeTypecastBehavior::resetOldAttributes() causes "class has no attribute named" InvalidArgumentException

### PR DESCRIPTION
**Pull request by uaoleg**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #19856

### CODE REVIEW

File: framework/db/BaseActiveRecord.php [hunk 1]
```diff
@@ -576,13 +576,24 @@ public function getOldAttribute($name)
...
         }
     }
 
+    /**
+     * Returns if the old named attribute can be set.
+     * @param string $name the attribute name
+     * @return bool whether the old attribute can be set
+     * @see setOldAttribute()
+     */
+    public function canSetOldAttribute($name): bool
```
Comment (samdark) [inline framework/db/BaseActiveRecord.php:592]:
```suggestion
    public function canSetOldAttribute($name)
```

**Comment by samdark** [comment]
@uaoleg please don't forget a line for CHANGELOG.

**Comment by uaoleg** [comment]
@samdark done

File: framework/db/BaseActiveRecord.php [hunk 2]
```diff
@@ -576,13 +576,24 @@ public function getOldAttribute($name)
...
         }
     }
 
+    /**
+     * Returns if the old named attribute can be set.
+     * @param string $name the attribute name
+     * @return bool whether the old attribute can be set
+     * @see setOldAttribute()
+     */
+    public function canSetOldAttribute($name)
```
Comment (SOHELAHMED7) [inline framework/db/BaseActiveRecord.php:592]: doesn't this method require unit test?

Comment (samdark) [reply framework/db/BaseActiveRecord.php:592]: If it's public — yes.

**Comment by samdark** [comment]
Thanks.

**Pull request merged**